File: LeetCode/153寻找旋转排序数组中的最小值.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int len = nums.size();
        if (len == 1) return nums[0];
        int left = 0, right = len - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[0] <= nums[mid] && nums[mid] > nums[len - 1]) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return nums[left];
    }
};

File: LeetCode/153寻找旋转排序数组中的最小值.go
func findMin(nums []int) int {
	size := len(nums)
	if size == 1 {
		return nums[0]
	}
	left, right := 0, size-1
	for left <= right {
		mid := left + (right-left)/2
		if nums[0] <= nums[mid] && nums[mid] > nums[size-1] {
			left = mid + 1
		} else {
			right = mid - 1
		}
	}
	return nums[left]
}